docs/articles/documentation/test-api/selecting-page-elements/examples-of-working-with-page-dom.md

```diff
@@ -0,0 +1,132 @@
+
+    // do something wuth clientWidth
+});
+```
+
+If you need to access an element property not included in the selector's API, request it explicitly by using the [addCustomDOMProperties](selectors/extending-selectors.md) method.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `https://devexpress.github.io/testcafe/example/`;
+
+test('Check Label HTML', async t => {
+    const label = Selector('label').addCustomDOMProperties({
+        innerHTML: el => el.innerHTML
+    });
+
+    await t.expect(label.innerHTML).contains('input type="checkbox" name="remote"');
+});
+```
+
+## Getting a Page Element Using Custom Logic
+
+Sometimes CSS selectors are not powerful enough to identify the required page element.
+In this instance, you can introduce a function that picks the desired element.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `https://devexpress.github.io/testcafe/example/`;
+
+test('My test', async t => {
+    const checkBoxesStartingWithR = Selector(() => {
+        var checkboxes     = document.querySelectorAll('input[type=checkbox]');
+        var targetElements = [];
+
+        checkboxes.forEach(node => {
+            if(node.labels[0].textContent.startsWith(' R'))
+                targetElements.push(node);
+        });
+
+        return targetElements;
+    });
+
+    await t.click(checkBoxesStartingWithR.nth(0));
+});
+```
+
+## Checking if an Element is Available
+
+Generally speaking, introducing conditions in tests is not considered good practice because it indicates that your tests are non-deterministic.
+The tested website should guarantee that the test writer knows the page state at any moment. If it is so, you need no conditions in test code.
+
+However, in practice, things are usually a bit different. Many websites contain elements that may be invisible or non-existent at times.
+In this instance, it may be a good idea to check the element availability before taking actions on it.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `https://devexpress.github.io/testcafe/example/`;
+
+test('My test', async t => {
+    const element = Selector('#developer-name');
+
+    if(await element.exists && await element.visible) {
+        await t.typeText(element, 'Peter Parker');
+    }
+
+    // ...
+});
+```
+
+## Enumerating Elements Identified by a Selector
+
+Another common case is creating a selector that matches several elements to perform certain actions using all of them.
+
+The following example clicks through a number of check boxes on the example page.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `https://devexpress.github.io/testcafe/example/`;
+
+test('My test', async t => {
+    const checkboxes = Selector('legend').withText('Which features are important to you:').parent(0).find('input');
+
+    for(let i = 0; i < await checkboxes.count; i++) {
```

Comment:
Small performance optiomizatin:
Now checkboxes count will calculated on each step of loop.
To prevent it, store a checkboxes count to the separate variable.

Reply:
we can use `for...of` here